soh/src/code/z_map_exp.c

```diff
@@ -851,26 +884,33 @@ void Minimap_Draw(PlayState* play) {
+                        s16 distFromCenter = (R_OW_MINIMAP_X + (gMapData->owMinimapWidth[mapIndex] / 2)) - (origX + (iconSize / 2));
+                        s16 mirrorOffset = distFromCenter * 2 + (iconSize / 2) - 2;
+                        s16 newX = origX + (CVarGetInteger("gMirroredWorld", 0) ? mirrorOffset : 0);
+
                         // The game authentically uses larger negative values for the entrance icon Y pos value. Normally only the first 12 bits
                         // would be read when the final value is passed into `gSPTextureRectangle`, but our cosmetic hud placements requires using
                         // `gSPWideTextureRectangle` which reads the first 24 bits instead. This caused the icon to be placed off screen.
                         // To address this, we take only the first 10 bits (which are later left-shifted by 2 to get our final 12 bits)
                         // to fix the entrance icon position when used with `gSPWideTextureRectangle`
                         s16 newY = gMapData->owEntranceIconPosY[sEntranceIconMapIndex] & 0x3FF;
-                        s16 PosX = gMapData->owEntranceIconPosX[sEntranceIconMapIndex] + X_Margins_Minimap;
 
-                        s16 entranceX = OTRGetRectDimensionFromRightEdge(PosX);
+                        s16 entranceX = OTRGetRectDimensionFromRightEdge(newX + X_Margins_Minimap);
                         s16 entranceY = newY + Y_Margins_Minimap;
                         if (CVarGetInteger("gMinimapPosType", 0) != 0) {
                             entranceY = newY + CVarGetInteger("gMinimapPosY", 0) + Y_Margins_Minimap;
                             if (CVarGetInteger("gMinimapPosType", 0) == 1) { // Anchor Left
                                 if (CVarGetInteger("gMinimapUseMargins", 0) != 0) {X_Margins_Minimap = Left_MM_Margin;};
-                                entranceX = OTRGetRectDimensionFromLeftEdge(PosX + CVarGetInteger("gMinimapPosX", 0));
+                                entranceX = OTRGetRectDimensionFromLeftEdge(newX + X_Margins_Minimap + CVarGetInteger("gMinimapPosX", 0));
                             } else if (CVarGetInteger("gMinimapPosType", 0) == 2) { // Anchor Right
                                 if (CVarGetInteger("gMinimapUseMargins", 0) != 0) {X_Margins_Minimap = Right_MM_Margin;};
-                                entranceX = OTRGetRectDimensionFromRightEdge(PosX + CVarGetInteger("gMinimapPosX", 0));
+                                entranceX = OTRGetRectDimensionFromRightEdge(newX + X_Margins_Minimap + CVarGetInteger("gMinimapPosX", 0));
                             } else if (CVarGetInteger("gMinimapPosType", 0) == 3) { // Anchor None
-                                entranceX = PosX + CVarGetInteger("gMinimapPosX", 0);
+                                entranceX = newX + X_Margins_Minimap + CVarGetInteger("gMinimapPosX", 0);
                             }
```

Comment:
Any particular reason why not keep `PosX` variable, set it equal to `newX + X_Margins_Minimap`, and keep all those calls to `PosX` from the original code?

Reply:
If you notice for Anchor left and Anchor right scenarios, `X_Margins_Minimap` can be overwritten with a new value, so it has to be reapplied. Technically this was not causing a problem before as higher up in this function, similar logic runs, so the `X_Margins_Minimap` would have already been overridden in this function scope. I could change it back, but really I think the hud placement logic could use a refactor overall.

Reply:
OK I think it makes sense to leave it then, thanks for the explanation.